plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -224,6 +229,18 @@ private CompletableFuture<ConnectorSplitBatch> getNextBatchInternal(int maxSize)
                     .completeOnTimeout(EMPTY_BATCH, timeLeft, MILLISECONDS);
         }
 
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
you may want to store this future somewhere and cancel on close()

Reply:
Yeah, cancellation needs to happen in the event `IcebergSplitSource#close()` is called. I'll wire that together in the next revision.